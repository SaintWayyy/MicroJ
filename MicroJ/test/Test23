/* This test for static function in class. */
class Animal {
    private static string species;
    public int age;
    double size;
    static string name;
    constructor Animal() {}
    constructor Animal(int age) {} /* the parameter has the same name as its field 'age */

    void eat(Animal self) {
        print("I eat meat");
    }
    static void sleep() {
        print("I'm sleeping");
    }
}

int main() {
    Animal lion : = new Animal(3);
    lion.eat();
    Animal.sleep();
    Animal.name = "King";
    print(Animal.name);
}