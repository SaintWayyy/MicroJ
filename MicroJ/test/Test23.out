"I eat meat"
"I'm sleeping"
"King"
